### PR TITLE
feat(package): switch to exact dependency versions

### LINKED ISSUE #843: Switch to exact dependency versions
Follow up of https://github.com/adobe/theblog/issues/416

This module depends on a lot of tiny dependencies, especially in the mdast / hast / dom / md transformations and manipulations. Those dependencies are referenced with the range operator which makes it hard to know which one of them breaks something or has an important impact.

Proposal: switch to exact dependency versions (like everywhere else)

### PR DESCRIPTION
**Pull request by kptdobe**
Fix #843 

- switch to exact dep
- update for each of them
- audit fix on the project (6 security fixes)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,44 +58,44 @@
...
+    "js-yaml": "3.14.0",
     "jsdom": "16.4.0",
-    "lodash": "^4.17.15",
-    "mdast-util-to-hast": "^9.0.0",
-    "mdast-util-to-string": "^1.0.6",
-    "micromatch": "^4.0.2",
-    "object-hash": "^2.0.0",
-    "path-to-regexp": "^6.1.0",
-    "property-information": "^5.2.2",
+    "lodash": "4.17.20",
```
Comment (tripodsan) [inline package.json:74]: do we need lodash? maybe we can replace it with only the package of the function that we need

File: package.json [hunk 2]
```diff
@@ -58,44 +58,44 @@
...
+    "mdast-util-to-hast": "9.1.1",
+    "mdast-util-to-string": "1.1.0",
+    "micromatch": "4.0.2",
+    "object-hash": "2.0.3",
+    "path-to-regexp": "6.1.0",
+    "property-information": "5.5.0",
     "proxyquire": "2.1.3",
```
Comment (tripodsan) [inline package.json:81]: why is proxyquire not in the `devDeps` ?

File: package.json [hunk 3]
```diff
@@ -58,44 +58,44 @@
...
-    "request": "^2.87.0",
-    "request-promise-native": "^1.0.7",
-    "retext": "^7.0.0",
-    "retext-smartypants": "^4.0.0",
-    "strip-markdown": "^3.1.0",
-    "unified": "^9.0.0",
-    "unist-util-map": "^2.0.0",
-    "unist-util-remove": "^2.0.0",
+    "remark-parse": "8.0.3",
+    "request": "2.88.2",
```
Comment (tripodsan) [inline package.json:83]: do we still need those?

File: package.json [hunk 4]
```diff
@@ -58,44 +58,44 @@
...
+    "request-promise-native": "1.0.9",
```
Comment (tripodsan) [inline package.json:84]: do we still need those?

**Review by trieloff — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (kptdobe) [reply package.json]: Moved to `devDeps`

File: package.json (hunk 1 above)
Comment (kptdobe) [reply package.json:73]: We need lodash in several places.

File: package.json (hunk 4 above)
Comment (kptdobe) [reply package.json]: `remark-parse` needed.
`request` is required by `jsdom` (even though it is not documented).
Removed `request-promise-native`.

File: package.json (hunk 3 above)
Comment (kptdobe) [reply package.json:81]: Yes, `request` is required by `jsdom`.

**Review by trieloff — APPROVED** [review]

**Comment by trieloff** [comment]
let's get a review from @rofe before this goes stale again

**Review by tripodsan — APPROVED** [review]

**Pull request merged**